app/Services/TagService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Services;
+
+use App\Repositories\TagRepository;
+use App\Http\Resources\TagResource;
+
+class TagService
+{
+    private $repo;
+    public function __construct(TagRepository $repo)
+    {
+        $this->repo = $repo;
+    }
```

Comment:
php 8.0 之後的版本可這樣寫

```suggestion
    public function __construct(TagRepository $repo)
    {
    }
```

前方也不用加 `private TagRepository $repo`

Reply:
要在建構子函式的括號內加上`public`(或`private`)

Reply:
個人習慣會用 private + readonly 避免被修改

```
public function __construct(private readonly TestServiceB $testServiceB)
{
}
```